src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -682,11 +682,21 @@ public function iShouldHaveProductsInTheCart(int $quantity, string $productName)
     }
 
     /**
+     * @Then /^my discount should be ("[^"]+")$/
      * @Then there should be no discount
      */
-    public function thereShouldBeNoDiscount(): void
+    public function myDiscountShouldBe(int $discount = 0): void
     {
-        Assert::same($this->responseChecker->getValue($this->client->getResponse(), 'orderPromotionTotal'), 0);
+        if ($this->sharedStorage->has('cart_token')) {
```

Comment:
Does this if makes sense? I would assume that cart_token is always set (as we always have cart declared).

Reply:
Yes, because it depends on context where it was started. If steps are all in checkout then they are using `$this->client->getResponse()` but some steps from cartContext are finished here and those steps are using `sharedStorage`.